### PR TITLE
Update plugin to use ip stack api

### PR DESCRIPTION
**Pull request by paigekato**
Tweak [Luke Holder's craft geo plugin](https://github.com/lukeholder/craft-geo) to use [IP Stack API](https://ipstack.com/) to return if user is in the EU.

### CODE REVIEW

File: geo/GeoPlugin.php [hunk 1]
```diff
@@ -5,7 +5,12 @@ class GeoPlugin extends BasePlugin
 {
     public function getName()
     {
-         return Craft::t('Geo');
+         return Craft::t('IP Stack');
```
Comment (dstello) [inline geo/GeoPlugin.php]: I think we can leave this as Geo?

File: geo/services/Geo_LocationService.php [hunk 2]
```diff
@@ -3,163 +3,89 @@
...
+			craft()->cache->add("craft.geo." . $ip,json_encode($data),$cacheTime);
+		}
+
+		return $data;
+	}
+
+	public function getIsEu()
+	{
+		$isEu = false;
+		$isEuUserCookie = 'flow_eu_user';
```
Comment (dstello) [inline geo/services/Geo_LocationService.php]: let's namespace this to the plugin and not to `flow`, `craft_geo_eu_user`

File: geo/services/Geo_LocationService.php [hunk 3]
```diff
@@ -3,163 +3,89 @@
...
-            return $anonymizedIp;
-        } else {
-            // Not a valid IP
-            return $ip;
-        }
-    }
-
-}
+
+	public function getIpData($doCacheData)
```
Comment (dstello) [inline geo/services/Geo_LocationService.php]: if we aren't going to update this as well let's just remove it

Comment (dstello) [reply geo/services/Geo_LocationService.php]: or update it to use the cookie method

File: geo/GeoPlugin.php [hunk 4]
```diff
@@ -8,18 +8,23 @@ public function getName()
          return Craft::t('Geo');
     }
 
+    public function getDescription()
+    {
+         return 'A tweaked version of geo cache by Luke Holder, to use a new IP API, IP Stack:https://ipstack.com';
```
Comment (dstello) [inline geo/GeoPlugin.php]: "geo cache" ?

Comment (paigekato) [reply geo/GeoPlugin.php]: geo caching

File: geo/services/Geo_LocationService.php [hunk 5]
```diff
@@ -3,163 +3,78 @@
...
+			'ip'=>$data->ip,
+			'country_code'=>$data->country_code,
+			'is_eu'=>$data->location->is_eu,
+			'cached'=>true
+		);
+		
+		$isEu = $data['is_eu'] === true ? 'true' : 'false';
+		setcookie($isEuUserCookie, $isEu);
+
+		if($doCacheData){
```
Comment (dstello) [inline geo/services/Geo_LocationService.php]: i think we can remove this? and $doCacheData?

File: geo/services/Geo_LocationService.php [hunk 6]
```diff
@@ -3,163 +3,78 @@
...
+		}
+		
+		$data = array(
+			'ip'=>$data->ip,
+			'country_code'=>$data->country_code,
+			'is_eu'=>$data->location->is_eu,
+			'cached'=>true
+		);
+		
+		$isEu = $data['is_eu'] === true ? 'true' : 'false';
```
Comment (dstello) [inline geo/services/Geo_LocationService.php]: i think we want to save $data in to the cookie and then parse EU out from the cookie
it's a little weird to have this "isEu" cookie in a function that returns an array of data.

**Pull request closed without merging**